### PR TITLE
refactored chatbox component, added photo url to users and chat messages, added tooltips

### PR DESCRIPTION
**Pull request by kawyua**
refactored and removed unused variables of chatbox component 
     - we had so many unused variables and information was being duplicated with roomnames and selectedchatroom when it was being used the same way.
  
added photo avatar and email tooltip to users in userlist 

added photo avatar in chat and display name as tooltip. This is dependent on the userlist of the chatroom
    - this required making a function call match userid of the chat message to a userid in userlist which             
    i made test functions for.

testtest

edit: after merging with hongzi's ui improvement 2
waitwhat

### CODE REVIEW

**Comment by Hongzred** [comment]
checked out locally and it looks as described and looks good.

File: src/app/chatbox/chatbox.component.html [hunk 1]
```diff
@@ -110,8 +101,8 @@ <h2>
         <img
           *ngIf="event.type == 'text'"
           matListAvatar
-          matTooltip="{{ event.from === this.userInfo.uid ?  userInfo.displayName : this.selectedConversation.name}}"
-          src="https://randomuser.me/api/portraits/thumb/lego/{{ event.from.charCodeAt(0) % 10 }}.jpg"
+          [matTooltip]="getUserName(event.from)"
```
Comment (Hongzred) [inline src/app/chatbox/chatbox.component.html]: good to convert them into functions, which make codes clearer. but did you write test for these functions yet?

File: src/app/chatbox/chatbox.component.ts [hunk 2]
```diff
@@ -405,6 +338,40 @@ export class ChatboxComponent implements OnInit, AfterViewChecked {
     }
   }
 
+  /**
```
Comment (Hongzred) [inline src/app/chatbox/chatbox.component.ts]: good documentation here and it looks like youve shorten the documentation for the variables

File: src/app/chatbox/chatbox.component.html (hunk 1 above)
Comment (kawyua) [reply src/app/chatbox/chatbox.component.html]: yes! its in the changes to chatbox.component.spec.ts files the test cases are there for the two functions.

```
it('calling getUserName() SHOULD return displayName IF valid userID', () => {
    componentUnderTest.userListEvents = USERS_IN_SELECTED_CHATROOM;
    expect(componentUnderTest.getUserName('userID1')).toEqual(USERS_IN_SELECTED_CHATROOM[0].displayName);
    expect(componentUnderTest.getUserName('userID2')).toEqual(USERS_IN_SELECTED_CHATROOM[1].displayName);
    expect(componentUnderTest.getUserName('anyuserstring')).toEqual('User is not in this channel');
  });

  it('calling getPhotoURL() SHOULD return photoURL IF valid userID', () => {
    componentUnderTest.userListEvents = USERS_IN_SELECTED_CHATROOM;
    expect(componentUnderTest.getPhoto('userID1')).toEqual(USERS_IN_SELECTED_CHATROOM[0].photoURL);
    expect(componentUnderTest.getPhoto('userID2')).toEqual(USERS_IN_SELECTED_CHATROOM[1].photoURL);
    expect(componentUnderTest.getPhoto('anyuserstring')).toEqual('http://bit.ly/chitty-ghost');
  });
```

**Pull request merged**